lua/luasnip/extras/snippet_list.lua

```diff
@@ -0,0 +1,96 @@
+local available = require("luasnip").available
+
+local function snip_info(snippet)
+	return {
+		name = snippet.name,
+		trigger = snippet.trigger,
+		description = snippet.dscr,
+		wordTrig = snippet.wordTrig and true or false,
+		regTrig = snippet.regTrig and true or false,
+		docstring = snippet:get_docstring(),
+	}
+end
+
+local function get_name()
+	return "LuaSnip://Snippets"
+end
+
+local win_opts = { foldmethod = "indent" }
+local buf_opts = { filetype = "lua" }
+
+local function set_win_opts(win, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_win_set_option(win, opt, val)
+	end
+end
+
+local function set_buf_opts(buf, opts)
+	for opt, val in pairs(opts) do
+		vim.api.nvim_buf_set_option(buf, opt, val)
+	end
+end
+
+local function make_scratch_buf(buf)
+	local opts = {
+		buftype = "nofile",
+		bufhidden = "wipe",
+		buflisted = false,
+		swapfile = false,
+		modified = false,
+		modeline = false,
+	}
+
+	set_buf_opts(buf, opts)
+end
+
+local function default_open(opts)
+	opts = opts or {}
+	opts.win_opts = opts.win_opts or win_opts
+	opts.buf_opts = opts.buf_opts or buf_opts
+	opts.get_name = opts.get_name or get_name
+
+	return function(printer_result)
+		-- create and open buffer on right vertical split
+		vim.cmd("botright vnew")
+
+		-- get buf and win handle
+		local buf = vim.api.nvim_get_current_buf()
+		local win = vim.api.nvim_get_current_win()
+
+		-- make scratch buffer
+		vim.api.nvim_buf_set_name(buf, opts.get_name())
```

Comment:
Ah, I think the `buf` passed to `opts.get_name` was lost somewhere :sweat_smile:

Reply:
Oh yeah oops, the default get_name might not use it but it might be useful for users that write their own get_name function haha. Added it back in.

Reply:
Perfect :ok_hand: :D